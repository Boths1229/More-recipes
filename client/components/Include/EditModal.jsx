import React from 'react';

const EditModal = (props) => {
  const { value, onChange, editRecipe, recipeId, saveImageToCloud } = props;
  return (
    <div
      className="modal fade"
      id="exampleModal1"
      tabIndex="-1"
      role="dialog"
      aria-labelledby="exampleModalLabel"
      aria-hidden="true"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title" id="exampleModalLabel">
              Update recipe
            </h5>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body">
            <form>
              <div className="form-group">
                <label htmlFor="recipient-name" className="col-form-label">
                  Name
                </label>
                <input
                  name="name"
                  onChange={onChange}
                  value={value.name}
                  type="text"
                  className="form-control"
                  id="recipient-name"
                />
              </div>
              <div className="form-group">
                <label htmlFor="message-text" className="col-form-label">
                  description
                </label>
                <textarea
                  name="description"
                  onChange={onChange}
                  value={value.description}
                  className="form-control"
                  id="message-text"
                />
              </div>
              <div className="form-group">
                <label htmlFor="ingredient" className="col-form-label black">
                  Ingredient &#58;
                </label>
                <input
                  id="ingredient"
                  type="text"
                  name="ingredient"
                  onChange={onChange}
                  value={value.ingredient}
                  className="form-control"
                />
              </div>
              <div className="form-group">
                <input
                  type="file"
                  name="image"
                  accept="image/*"
                  onChange={saveImageToCloud}
                />
              </div>
            </form>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="btn btn-secondary"
              data-dismiss="modal"
            >
              Close
            </button>
            <button
              type="button"
              className="btn btn-orange"
              onClick={() => editRecipe(recipeId)}
            >
              Update recipe
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default EditModal;
