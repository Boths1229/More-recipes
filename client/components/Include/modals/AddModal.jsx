import React from 'react';
import PropTypes, { any } from 'prop-types';

/**
 * @description functional component to render modal for adding recipes
 *
 * @param {props} props
 *
 * @returns {JSX} return JSX
 */
const AddModal = (props) => {
  const {
    onChange, value, saveImageToCloud, addRecipe, recipeImage
  } = props;
  const {
    name, description, ingredient, responseMessage
  } = value;
  return (
    <div
      className="modal fade"
      id="exampleModal"
      tabIndex="-1"
      role="dialog"
      aria-labelledby="exampleModalLabel"
      aria-hidden="true"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header black">
            <h5 className="modal-title black" id="exampleModalLabel">
              Add new recipe
            </h5>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body">
            <form>
              <div className="form-group">
                <div className="error-message">
                  { responseMessage }
                </div>
                <label
                  htmlFor="recipient-name"
                  className="col-form-label black"
                >
                  Name &#58;
                </label>
                <input
                  name="name"
                  onChange={onChange}
                  value={name}
                  type="text"
                  className="form-control"
                  id="recipient-name"
                />
              </div>
              <div className="form-group">
                <label htmlFor="message-text" className="col-form-label black">
                  Description &#58;
                </label>
                <textarea
                  name="description"
                  onChange={onChange}
                  value={description}
                  className="form-control"
                  id="message-text"
                />
              </div>
              <div className="form-group">
                <label htmlFor="ingredient" className="col-form-label black">
                  Ingredient &#58;
                </label>
                <input
                  id="ingredient"
                  placeholder="separate each ingredient with a comma"
                  type="text"
                  name="ingredient"
                  onChange={onChange}
                  value={ingredient}
                  className="form-control"
                />
              </div>
              <div className="form-group img-text">
                <input
                  type="file"
                  name="image"
                  accept="image/*"
                  onChange={saveImageToCloud}
                />
              </div>
            </form>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="btn btn-secondary"
              data-dismiss="modal"
            >
              Close
            </button>
            <button
              type="button"
              disabled={!recipeImage ? 'disabled' : null}
              className="btn btn-orange"
              onClick={addRecipe}
            >
              Add recipe
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

AddModal.propTypes = {
  addRecipe: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  saveImageToCloud: PropTypes.func.isRequired,
  value: PropTypes.objectOf(any).isRequired
};


export default AddModal;
