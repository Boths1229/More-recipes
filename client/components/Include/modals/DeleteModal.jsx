import React from 'react';
import PropTypes from 'prop-types';

/**
 * @description functional component to render the delete recipe confirmation modal
 *
 * @param {object} props
 *
 * @returns {JSX} JSX
 */
const DeleteModal = (props) => {
  const { deleteRecipe, recipeId } = props;
  return (
    <div
      className="modal fade"
      id={`deleteModal-${recipeId}`}
      tabIndex="-1"
      role="dialog"
      aria-labelledby="exampleModalLabel"
      aria-hidden="true"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title" id="exampleModalLabel1">
              Delete recipe
            </h5>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body">
            This action is irrevesible, pls process with caution
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="btn btn-secondary"
              data-dismiss="modal"
            >
              Close
            </button>
            <button
              type="button"
              onClick={() => deleteRecipe(recipeId)}
              className="btn btn-danger"
              data-dismiss="modal"
            >
              Delete recipe
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

DeleteModal.propTypes = {
  deleteRecipe: PropTypes.func.isRequired,
  recipeId: PropTypes.number.isRequired
};

export default DeleteModal;
