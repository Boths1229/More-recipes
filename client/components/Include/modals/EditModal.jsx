import React from 'react';
import PropTypes, { any } from 'prop-types';

/**
 * @description function component to render the edit recipe modal
 *
 * @param {object} props
 *
 * @returns {JSX} return JSX
 */
const EditModal = (props) => {
  const {
    recipeData, handleCloseModal, onChange, cardId, value, editRecipe, saveImageToCloud,
  } = props;
  const {
    id, name, description, ingredient
  } = recipeData;
  return (
    <div
      className="modal fade"
      id={cardId}
      tabIndex="-1"
      role="dialog"
      aria-labelledby="exampleModalLabel"
      aria-hidden="true"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title" id="exampleModalLabel">
              Update recipe
            </h5>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
              onClick={handleCloseModal}
            >
              <span
                aria-hidden="true"
              >&times;
              </span>
            </button>
          </div>
          <div className="modal-body">
            <form>
              <div className="form-group">
                <label htmlFor="recipient-name" id="recipient-name" className="col-form-label">
                  Name
                </label>
                <input
                  name="name"
                  onChange={onChange}
                  value={value.isChanged ? value.name : name}
                  type="text"
                  className="form-control"
                  id="recipient-name"
                />
              </div>
              <div className="form-group">
                <label htmlFor="message-text" className="col-form-label">
                  description
                </label>
                <textarea
                  name="description"
                  onChange={onChange}
                  value={value.isChanged ? value.description : description}
                  className="form-control"
                  id="message-text"
                />
              </div>
              <div className="form-group">
                <label htmlFor="ingredient" className="col-form-label black">
                  Ingredient &#58;
                </label>
                <input
                  id="ingredient"
                  type="text"
                  name="ingredient"
                  onChange={onChange}
                  value={value.isChanged ? value.ingredient : ingredient}
                  className="form-control"
                />
              </div>
              <div className="form-group">
                <input
                  type="file"
                  name="image"
                  accept="image/*"
                  onChange={saveImageToCloud}
                />
              </div>
            </form>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="btn btn-secondary"
              data-dismiss="modal"
              onClick={handleCloseModal}
            >
              Close
            </button>
            <button
              type="button"
              className="btn btn-orange"
              onClick={() => editRecipe(id, recipeData)}
            >
              Update recipe
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

EditModal.propTypes = {
  value: PropTypes.objectOf(any).isRequired,
  onChange: PropTypes.func.isRequired,
  editRecipe: PropTypes.func.isRequired,
  saveImageToCloud: PropTypes.func.isRequired,
  handleCloseModal: PropTypes.func.isRequired,
  recipeData: PropTypes.objectOf(any).isRequired,
  cardId: PropTypes.number.isRequired
};

export default EditModal;
