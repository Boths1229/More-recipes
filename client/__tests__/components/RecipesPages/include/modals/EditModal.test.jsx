import React from 'react';
import { shallow } from 'enzyme';
import { EditModal } from '../../../../../components/Include/modals/EditModal';
import mockData from '../../../../__mocks__/recipeData.json';

const {
  addRecipe
} = mockData.Recipes;

describe('Edit modal component', () => {
  it('matches snapshot', () => {
    const props = {
      editRecipeStatus: '',
    };
    const output = shallow(<EditModal {...props} />);
    expect(output).toMatchSnapshot();
  });
  describe('instance and lifecycle methods', () => {
    const props = {
      editRecipeAction: jest.fn(),
      saveImageToCloudAction: jest.fn(),
      image: '',
      editRecipeStatus: false,
    };
    // const state = {
    //   recipe: {
    //     id: 6,
    //     name: Coke,
    //     description: cook for 10mins,
    //     ingredient: bla,
    //     views: 0,
    //     image: rice,
    //     userId: 4,
    //     upvotes: 0,
    //     downvotes: 0,
    //     favorites: 1,
    //   }
    // };
    it('calls componentWillRecieveProps', () => {
      const wrapper = shallow(<EditModal {...props} />);
      const cwrp = jest.spyOn(wrapper.instance(), 'componentWillReceiveProps');
      wrapper.setProps({ image: 'image' });
      expect(cwrp).toHaveBeenCalled();
    });
    it('handleEditRecipeChanges', () => {
      const wrapper = shallow(<EditModal {...props} />);
      wrapper.setState({ recipe: null });
      const setState = jest.spyOn(wrapper.instance(), 'setState');
      wrapper.instance().editChangeHandler({ target: { name: 'description', value: '' } });
      expect(setState).toHaveBeenCalledWith({
        recipe: {
          description: '', image: '', ingredient: '', name: ''
        }
      });
    });
    it('handles edit recipe submission', () => {
      const wrapper = shallow(<EditModal {...props} />);
      wrapper.setState({ recipe: addRecipe });
      wrapper.instance().onSubmit();
      expect(props.editRecipeAction).toBeCalled();
    });
    it('handles saveImageToCloud', () => {
      const wrapper = shallow(<EditModal {...props} />);
      wrapper.instance().saveImageToCloud({ target: { files: [{ image: 'recipe' }] } });
      expect(props.saveImageToCloudAction).toBeCalled();
    });
  });
});
