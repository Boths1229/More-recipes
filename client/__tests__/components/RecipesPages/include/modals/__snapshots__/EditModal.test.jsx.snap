// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit modal component matches snapshot 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalLabel"
  className="modal fade"
  id="editModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLabel"
        >
          Update recipe
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <form>
          <div
            className="form-group"
          >
            <div
              className="error-message"
            />
            <label
              className="col-form-label"
              htmlFor="recipient-name"
              id="recipient-name"
            >
              Name
            </label>
            <input
              className="form-control name"
              id="recipient-name"
              name="name"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label"
              htmlFor="message-text"
            >
              description
            </label>
            <textarea
              className="form-control"
              id="message-text"
              name="description"
              onChange={[Function]}
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="ingredient"
            >
              Ingredient :
            </label>
            <input
              className="form-control"
              id="ingredient"
              name="ingredient"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              accept="image/*"
              name="image"
              onChange={[Function]}
              type="file"
            />
          </div>
        </form>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Close
        </button>
        <button
          className="btn btn-orange editmodal"
          onClick={[Function]}
          type="button"
        >
          Update recipe
        </button>
      </div>
    </div>
  </div>
</div>
`;
